client/src/components/RelatedItems/RelatedList/CardImage.jsx

```diff
@@ -1,55 +1,65 @@
 import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
-import axios from 'axios';
-import { useGlobalContext } from '../../../contexts/GlobalStore';
+import PropTypes from 'prop-types';
+import ComparisonModal from './ComparisonModal';
 
-function CardImage(imageObj) {
-  // console.log('CardImage', imageObj);
-  const [image, setImage] = useState(imageObj);
-  const {
-    outfits, setOutfits,
-  } = useGlobalContext();
+function CardImage({ imageInfo, details }) {
+  const [image, setImage] = useState(imageInfo);
+  const [modal, setModal] = useState(false);
+  const defaultImage = 'https://www.escapeauthority.com/wp-content/uploads/2116/11/No-image-found.jpg';
   useEffect(() => {
-    setImage(imageObj);
-  }, [imageObj]);
-  function handleAdd() {
-    const newOutfit = imageObj;
-    // newOutfit.thumbnail = image;
-    // Note: Need to use below syntax for component to re-render properly
-    const tempArray = [...outfits, newOutfit];
-    // tempArray.push(newOutfit);
-    // console.log(tempArray);
-    setOutfits(tempArray);
-    // console.log('Outfit list', outfits);
+    setImage(imageInfo);
+  }, [imageInfo]);
+  function handleModal() {
+    if (modal) {
+      setModal(false);
+    } else {
+      setModal(true);
+    }
   }
 
   return (
-    <div>
-      <ImageCard src={image.imageInfo.results[0].photos[0].thumbnail_url} alt="RelatedProductImage" />
+    <Outline onClick={(e) => { if (modal) { e.stopPropagation(); handleModal(); } }}>
```

Comment:
It is not very obvious how to close the modal, you can instead pass the setModal function to the modal as a prop, and use it within the modal